Compilation fails with error in pdf/XPDFRenderer.cpp

Version 1.6.1-5 fails to compile on Arch linux with the following error:

```src/pdf/XPDFRenderer.cpp: In constructor ‘XPDFRenderer::XPDFRenderer(const QString&, bool)’:
src/pdf/XPDFRenderer.cpp:99:74: error: no matching function for call to ‘PDFDoc::PDFDoc(GooString*, int, int, int)’
   99 |     mDocument = new PDFDoc(new GooString(filename.toLocal8Bit()), 0, 0, 0); // the filename GString is deleted on PDFDoc desctruction
      |                                                                          ^
In file included from src/pdf/XPDFRenderer.h:51,
                 from src/pdf/XPDFRenderer.cpp:30:
/usr/include/poppler/PDFDoc.h:371:5: note: candidate: ‘PDFDoc::PDFDoc()’
  371 |     PDFDoc();
      |     ^~~~~~
/usr/include/poppler/PDFDoc.h:371:5: note:   candidate expects 0 arguments, 4 provided
/usr/include/poppler/PDFDoc.h:139:14: note: candidate: ‘PDFDoc::PDFDoc(BaseStream*, const std::optional<GooString>&, const std::optional<GooString>&, void*, const std::function<void()>&)’
  139 |     explicit PDFDoc(BaseStream *strA, const std::optional<GooString> &ownerPassword = {}, const std::optional<GooString> &userPassword = {}, void *guiDataA = nullptr, const std::function<void()> &xrefReconstructedCallback = {});
      |              ^~~~~~
/usr/include/poppler/PDFDoc.h:139:33: note:   no known conversion for argument 1 from ‘GooString*’ to ‘BaseStream*’
  139 |     explicit PDFDoc(BaseStream *strA, const std::optional<GooString> &ownerPassword = {}, const std::optional<GooString> &userPassword = {}, void *guiDataA = nullptr, const std::function<void()> &xrefReconstructedCallback = {});
      |                     ~~~~~~~~~~~~^~~~
In file included from src/pdf/XPDFRenderer.h:51,
                 from src/pdf/XPDFRenderer.cpp:30:
/usr/include/poppler/PDFDoc.h:132:14: note: candidate: ‘PDFDoc::PDFDoc(std::unique_ptr<GooString>&&, const std::optional<GooString>&, const std::optional<GooString>&, void*, const std::function<void()>&)’
  132 |     explicit PDFDoc(std::unique_ptr<GooString> &&fileNameA, const std::optional<GooString> &ownerPassword = {}, const std::optional<GooString> &userPassword = {}, void *guiDataA = nullptr,
      |              ^~~~~~
/usr/include/poppler/PDFDoc.h:132:50: note:   no known conversion for argument 1 from ‘GooString*’ to ‘std::unique_ptr<GooString>&&’
  132 |     explicit PDFDoc(std::unique_ptr<GooString> &&fileNameA, const std::optional<GooString> &ownerPassword = {}, const std::optional<GooString> &userPassword = {}, void *guiDataA = nullptr,
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
```
inkscape had a similar issue a few days ago that looks related: https://gitlab.com/inkscape/inbox/-/issues/6563

I have poppler 22.03.0-1 (upgraded 6 mar 2022).

--- a/src/pdf/XPDFRenderer.cpp
+++ b/src/pdf/XPDFRenderer.cpp
@@ -95,6 +95,8 @@ XPDFRenderer::XPDFRenderer(const QString &filename, bool importingFile)
     }
 #ifdef USE_XPDF
     mDocument = new PDFDoc(new GString(filename.toLocal8Bit()), 0, 0, 0); // the filename GString is deleted on PDFDoc desctruction
+#elif POPPLER_VERSION_MAJOR > 22 || (POPPLER_VERSION_MAJOR == 22 && POPPLER_VERSION_MINOR >= 3)
+    mDocument = new PDFDoc(std::make_unique<GooString>(filename.toLocal8Bit()));
 #else
     mDocument = new PDFDoc(new GooString(filename.toLocal8Bit()), 0, 0, 0); // the filename GString is deleted on PDFDoc desctruction
 #endif

--- a/src/pdf/pdf.pri
+++ b/src/pdf/pdf.pri
@@ -1,3 +1,4 @@
+CONFIG += c++17
 
 HEADERS      += src/pdf/GraphicsPDFItem.h \
                 src/pdf/PDFRenderer.h \

